src/libxtp/jobcalculators/qmmm.cc

```diff
@@ -225,36 +239,55 @@ void QMMM::WriteJobFile(const Topology& top) {
 
   ofs << "<jobs>" << std::endl;
   Index jobid = 0;
-  std::string regionname = getFirstRegionName();
-  bool hasqm = hasQMRegion();
-  for (const Segment& seg : top.Segments()) {
-    for (const QMState& state : states_) {
-
-      std::string marker = std::to_string(seg.getId()) + ":" + state.ToString();
-      std::string tag = seg.getType() + "_" + marker;
-
-      tools::Property Input;
-      tools::Property& pInput = Input.add("input", "");
-      pInput.add("site_energies", marker);
-      tools::Property& regions = pInput.add("regions", "");
-      tools::Property& region = regions.add(regionname, "");
-      region.add("id", "0");
-      if (hasqm) {
-        region.add("state", state.ToString());
+  if (all_segments_) {
+    for (const Segment& seg : top.Segments()) {
+      for (const QMState& state : states_) {
+        Job job = createJob(seg, state, jobid);
+        job.ToStream(ofs);
+        jobid++;
+      }
+    }
+  } else {
+    for (Index segID : segments_) {
+      const Segment& seg = top.Segments()[segID];
+      for (const QMState& state : states_) {
+        Job job = createJob(seg, state, jobid);
+        job.ToStream(ofs);
+        jobid++;
       }
-      region.add("segments", marker);
-      Job job(jobid, tag, Input, Job::AVAILABLE);
-      job.ToStream(ofs);
-      jobid++;
     }
   }
 
   ofs << "</jobs>" << std::endl;
   ofs.close();
   std::cout << std::endl
-            << "... ... In total " << jobid + 1 << " jobs" << std::flush;
+            << "... ... In total " << jobid << " jobs" << std::flush;
```

Comment:
why did you remove the `+1` ?

Reply:
because it is wrong, at the end of the for loop above you end with `jobid++` so it always prints one job to much in the summary

Reply:
good point. :)